### PR TITLE
[6150] Shared component path is not updated in parent renamed

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6150

### CODE REVIEW

**Review by rart — COMMENTED** [review]
Let's review together.

File: ui/app/src/components/DependenciesDialog/DependenciesDialog.tsx [hunk 1]
```diff
@@ -46,8 +46,7 @@ export const dependenciesDialogStyles = makeStyles()((theme) => ({
   dependenciesList: {
     backgroundColor: theme.palette.background.paper,
     padding: 0,
-    borderRadius: '5px 5px 0 0',
-    overflowY: 'auto'
+    borderRadius: '5px 5px 0 0'
```
Comment (rart) [inline ui/app/src/components/DependenciesDialog/DependenciesDialog.tsx:49]: Make sure to test with large list of dependencies.

File: ui/app/src/components/RenameContentDialog/RenameContentDialogContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      (name: string) => {
+        checkPathExistence(
+          siteId,
+          `${ensureSingleSlash(`${path}/${name}`)}${isPage ? '/index.xml' : '.xml'}`
+        ).subscribe((exists) => {
+          setItemExists(name !== strippedValue && exists);
+        });
+      },
+      [path, siteId, strippedValue, isPage]
+    ),
```
Comment (rart) [inline ui/app/src/components/RenameContentDialog/RenameContentDialogContainer.tsx]: There's no need for callback memoizing, `useDebouncedInput` uses a ref internally.
```suggestion
    (name: string) => {
      checkPathExistence(
        siteId,
        `${ensureSingleSlash(`${path}/${name}`)}${isPage ? '/index.xml' : '.xml'}`
      ).subscribe((exists) => {
        setItemExists(name !== strippedValue && exists);
      });
    }
```

File: ui/app/src/components/RenameContentDialog/RenameContentDialogContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+  useEffect(() => {
+    refs.current.onNameUpdate$.next(name);
+  }, [name, refs]);
+
+  const onInputChanges = (newValue: string) => {
+    setName(newValue);
+    const newHasPendingChanges = newValue !== strippedValue;
+    onSubmittingAndOrPendingChange({ hasPendingChanges: newHasPendingChanges });
+  };
```
Comment (rart) [inline ui/app/src/components/RenameContentDialog/RenameContentDialogContainer.tsx]:
```suggestion
  const onInputChanges = (newValue: string) => {
    setName(newValue);
    refs.current.onNameUpdate$.next(name);
    const newHasPendingChanges = newValue !== strippedValue;
    onSubmittingAndOrPendingChange({ hasPendingChanges: newHasPendingChanges });
  };
```

File: ui/app/src/components/RenameDialogBody/RenameItemView.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  confirmBrokenReferences: boolean;
+  setConfirmBrokenReferences: (value: boolean) => void;
+  fetchingDependantItems: boolean;
+  error: AjaxError;
+  onRename: () => void;
+  onInputChanges: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  helperText?: ReactNode;
+}
+
+export function RenameItemView(props: RenameDialogBodyProps) {
```
Comment (rart) [inline ui/app/src/components/RenameDialogBody/RenameItemView.tsx]:
```suggestion
export interface RenameItemViewProps {
  name: string;
  disabled: boolean;
  newNameExists: boolean;
  dependantItems: DetailedItem[];
  isSubmitting: boolean;
  confirmBrokenReferences: boolean;
  setConfirmBrokenReferences: (value: boolean) => void;
  fetchingDependantItems: boolean;
  error: AjaxError;
  onRename: () => void;
  onInputChanges: (event: React.ChangeEvent<HTMLInputElement>) => void;
  helperText?: ReactNode;
}

export function RenameItemView(props: RenameItemViewProps) {
```

File: ui/app/src/utils/content.ts [hunk 5]
```diff
@@ -1021,6 +1021,10 @@ export function applyAssetNameRules(name: string, options?: { allowBraces: boole
   return name.replace(options?.allowBraces ? /[^a-zA-Z0-9-_{}.]/g : /[^a-zA-Z0-9-_.]/g, '').replace(/\.{1,}/g, '.');
 }
 
+export function applyContentNameRules(name: string): string {
```
Comment (rart) [inline ui/app/src/utils/content.ts:1028]:
```suggestion
/**
 * Please add content explaining more...
 */
export function applyContentNameRules(name: string): string {
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/DependenciesDialog/DependenciesDialog.tsx (hunk 1 above)
Comment (jvega190) [reply ui/app/src/components/DependenciesDialog/DependenciesDialog.tsx:49]: Per conversation we'll leave it without `overflowY`

File: ui/app/src/components/RenameContentDialog/RenameContentDialogContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  const siteId = useActiveSiteId();
+  const refs = useUpdateRefs({ onNameUpdate$: null });
+
+  refs.current.onNameUpdate$ = useDebouncedInput((name: string) => {
+    checkPathExistence(siteId, `${ensureSingleSlash(`${path}/${name}`)}${isPage ? '/index.xml' : '.xml'}`).subscribe(
+      (exists) => {
+        setItemExists(name !== strippedValue && exists);
+      }
+    );
+  }, 400);
```
Comment (rart) [inline ui/app/src/components/RenameContentDialog/RenameContentDialogContainer.tsx]: Don't really see the need for this as a ref.

```suggestion
  const refs = useUpdateRefs({

onNameUpdate$: useDebouncedInput((name: string) => {
    checkPathExistence(siteId, `${ensureSingleSlash(`${path}/${name}`)}${isPage ? '/index.xml' : '.xml'}`).subscribe(
      (exists) => {
        setItemExists(name !== strippedValue && exists);
      }
    );
  }, 400)

});
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/RenameContentDialog/RenameContentDialogContainer.tsx (hunk 6 above)
Comment (jvega190) [reply ui/app/src/components/RenameContentDialog/RenameContentDialogContainer.tsx]: Yeah not needed, was initially supposed to be used differently. Removing ref

**Review by rart — APPROVED** [review]

**Pull request merged**